internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -105,20 +110,82 @@ func (p *processRunner) build() (cmd *exec.Cmd, logCloser io.Closer, err error)
 		writer = w
 	}
 
-	cmd.Stdout = writer
-	cmd.Stderr = writer
+	var afterStartFuncs []func()
+
+	if p.opts.StdinFile != "" {
+		stdin, err := os.Open(p.opts.StdinFile)
+		if err != nil {
+			return commandWrapper{}, err
+		}
+
+		cmd.Stdin = stdin
+
+		afterStartFuncs = append(afterStartFuncs, func() {
+			stdin.Close() //nolint:errcheck
+		})
+	}
+
+	if p.opts.StdoutFile != "" {
+		stdout, err := os.OpenFile(p.opts.StdoutFile, os.O_WRONLY, 0)
+		if err != nil {
+			return commandWrapper{}, err
+		}
 
-	return cmd, w, nil
+		cmd.Stdout = stdout
+
+		afterStartFuncs = append(afterStartFuncs, func() {
+			stdout.Close() //nolint:errcheck
+		})
+	} else {
+		cmd.Stdout = writer
+	}
+
+	if p.opts.StderrFile != "" {
+		stderr, err := os.OpenFile(p.opts.StderrFile, os.O_WRONLY, 0)
+		if err != nil {
+			return commandWrapper{}, err
+		}
+
+		cmd.Stderr = stderr
+
+		afterStartFuncs = append(afterStartFuncs, func() {
+			stderr.Close() //nolint:errcheck
+		})
+	} else {
+		cmd.Stderr = writer
+	}
+
+	if p.opts.Ctty.IsPresent() {
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Setsid:  true,
+			Setctty: true,
+			Ctty:    1,
```

Comment:
should be the the value of `p.opts.Ctty` ?

Reply:
yep, my bad, fixed.